### DISCUSSION TITLE
HelmChart reuse

### DISCUSSION THREAD

**Discussion by aaovchinnikov** [fluxcd/flux2#3895] [General] [answered]
Hi, everyone!

Flux v2 is cool, but it seems to be some unclear spots here with HelmChart resource.

As I can see in [documentation](https://fluxcd.io/flux/components/source/helmcharts/), I can declare Flux `HelmChart` resource, but I don't see any place where to use it.

On the other hand when I create two `HelmRelease` resources that rely on same chart in `HelmRepository`, Flux creates two separate but in fact same `HelmChart` resources with template in `spec.chart` of HelmReleases.

It seems to be some kind of unnecessary API-resource duplication, while "hand-made" HelmChart object just can't be referenced from HelmReleases for reuse.

Am I missing something? Is there any way to avoid this `HelmChart` duplication? I have some kind of generic chart for deploying a lot of applications to cluster and listing such same `HelmChart` resources is bothering me.

**Comment by PoovenB** [comment, marked as answer]
You've probably moved on from this, but for anyone else search, the `HelmRelease` spec now supports a `chartRef`. See https://fluxcd.io/flux/components/helm/helmreleases/#chart-reference

Thus you are able to define the helm chart once, and reference that in as many helm releases as you'd like.

**Comment by aaovchinnikov** [comment]
Yes, that's definitely solves the question!
Many thanks!
